AdminCore.FsmWorkflow/FsmMachines/WorkflowFsmWfh.cs

```diff
@@ -0,0 +1,122 @@
+                                break;
+                        }
+                    }
+                })
+                .InternalTransition(LeaveResponseTrigger,
+                    (approvalState, responder, transition) => LeaveResponse(approvalState, responder))
+                .Permit(LeaveTriggersWfh.LeaveApproved, WfhState.LeaveApproved)
+                .Permit(LeaveTriggersWfh.LeaveRejected, WfhState.LeaveRejected)
+                .Permit(LeaveTriggersWfh.LeaveCancelled, WfhState.LeaveCancelled)
+                .Permit(LeaveTriggersWfh.AdminApprove, WfhState.LeaveApproved)
+                .Permit(LeaveTriggersWfh.AdminReject, WfhState.LeaveRejected)
+                .PermitReentry(LeaveTriggersWfh.EvaluateLeaveState);
+            
+            // Leave Approved
+            FsMachine.Configure(WfhState.LeaveApproved)
+                .SubstateOf(WfhState.LeaveRequestCompleted)
+                .OnEntry(LeaveApproved)
+                .Ignore(LeaveTriggersWfh.AdminReject)
+                .Ignore(LeaveTriggersWfh.AdminApprove)
+                .Ignore(LeaveTriggersWfh.LeaveApproved)
+                .Ignore(LeaveTriggersWfh.LeaveRejected)
+                .Ignore(LeaveTriggersWfh.LeaveCancelled)
+                .Ignore(LeaveTriggersWfh.LeaveResponded)
+                .Ignore(LeaveTriggersWfh.EvaluateLeaveState)
+                .Ignore(LeaveTriggersWfh.TeamLeadResponseReceived);
+            
+            // Leave Rejected
+            FsMachine.Configure(WfhState.LeaveRejected)
+                .SubstateOf(WfhState.LeaveRequestCompleted)
+                .OnEntry(LeaveRejected)
+                .Ignore(LeaveTriggersWfh.AdminReject)
+                .Ignore(LeaveTriggersWfh.AdminApprove)
+                .Ignore(LeaveTriggersWfh.LeaveApproved)
+                .Ignore(LeaveTriggersWfh.LeaveRejected)
+                .Ignore(LeaveTriggersWfh.LeaveCancelled)
+                .Ignore(LeaveTriggersWfh.LeaveResponded)
+                .Ignore(LeaveTriggersWfh.EvaluateLeaveState)
+                .Ignore(LeaveTriggersWfh.TeamLeadResponseReceived);
+            
+            // Leave Cancelled
+            FsMachine.Configure(WfhState.LeaveCancelled)
+                .SubstateOf(WfhState.LeaveRequestCompleted)
+                .OnEntry(LeaveCancelled)
+                .Ignore(LeaveTriggersWfh.AdminReject)
+                .Ignore(LeaveTriggersWfh.AdminApprove)
+                .Ignore(LeaveTriggersWfh.LeaveApproved)
+                .Ignore(LeaveTriggersWfh.LeaveRejected)
+                .Ignore(LeaveTriggersWfh.LeaveCancelled)
+                .Ignore(LeaveTriggersWfh.LeaveResponded)
+                .Ignore(LeaveTriggersWfh.EvaluateLeaveState)
+                .Ignore(LeaveTriggersWfh.TeamLeadResponseReceived);
+            
+            FsMachine.Activate();
+        }
+        
+        public override WorkflowFsmStateInfo FireLeaveResponded(EventStatuses approvalState, string responder)
```

Comment:
I strongly think this function has a potential to be reused between different workflows!

Reply:
Unfortunately not as the trigger enums used by the PTO and WFH workflow are different. LeaveTriggersPto vs LeaveTriggersWfh

Reply:
I don't think this will scale very well with enums... do we expect to add this function to every new workflow?? If the answer is yes then consider using classes instead of enums that implement a common interface and put this function in the abstract class.

Reply:
I pushed the changes for this, you can find them in FsmWorkflow.TriggerStateAccessor and in the WorkflowFsmPto and WorkflowFsmWfh FSMs.